alphametics/example.go

```diff
@@ -0,0 +1,200 @@
+package alphametics
+
+import "sync"
+
+const TestVersion = 1
+
+// Set this to true to debug the example. It enables several useful logging tools.
+const debugging = false
+
+type accumulator struct {
+	carry    int
+	usedNums uint16       // bit n denotes if digit n is taken
+	solution map[rune]int // partially filled solution map
+}
+
+type ipc struct {
+	resultChan chan map[rune]int
+	abortChan  chan struct{}
+	waitGroup  *sync.WaitGroup
+}
+
+func Solve(a, b, c string) map[rune]int {
+	aRunes, bRunes, cRunes := prepareRunes(a, b, c)
+
+	// Setup the machinery for solving in parallel.
+	resultChan := make(chan map[rune]int)
```

Comment:
It may also be helpful to resolve this w/o concurrency and then add it in if needed only after you have a clean solution.